src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -923,47 +955,52 @@ ov::pass::NopStridedSlice::NopStridedSlice() {
     register_matcher(m, matcher_pass_callback);
 }
 
-ov::pass::NopStridedSliceByShape::NopStridedSliceByShape() {
-    MATCHER_SCOPE(NopStridedSliceByShape);
-    auto slice = pattern::wrap_type<op::v8::Slice>();
+ov::pass::EliminateStridedSliceByShape::EliminateStridedSliceByShape() {
+    MATCHER_SCOPE(EliminateStridedSliceByShape);
 
     auto input = pattern::any_input();
-    auto begin_const = pattern::any_input();
-    auto end_const = pattern::any_input();
+    auto begin = pattern::any_input();
+    auto end = pattern::any_input();
     auto optional_stride_const = pattern::wrap_type<ov::op::v0::Constant>();
-    auto pattern = pattern::wrap_type<ov::op::v1::StridedSlice>({input, begin_const, end_const, optional_stride_const});
+    auto strided_slice = pattern::wrap_type<ov::op::v1::StridedSlice>({input, begin, end, optional_stride_const});
+
+    auto axes = pattern::any_input();
+    auto slice = pattern::wrap_type<ov::op::v8::Slice>({input, begin, end, optional_stride_const, axes});
+    auto pattern = std::make_shared<pattern::op::Or>(OutputVector{strided_slice, slice});
 
     ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
```

Comment:
Overall, I agree with the idea of this and other Elimination passes here. However, they seem complex, so I strongly recommend rewriting them using Symbolic infra in the future.